test/new-e2e/tests/containers/k8s_test.go

```diff
@@ -797,40 +800,77 @@ func (suite *k8sSuite) TestSBOM() {
+		if !assert.NotEmptyf(c, images, "No SBOM payload yet") {
 			return
 		}
 
-		images, err := suite.Fakeintake.FilterSBOMs(sbomID)
-		// Can be replaced by require.NoErrorf(…) once https://github.com/stretchr/testify/pull/1481 is merged
-		if !assert.NoErrorf(c, err, "Failed to query fake intake") {
+		images = lo.Filter(images, func(image *aggregator.SBOMPayload, _ int) bool {
+			return image.Status == sbom.SBOMStatus_SUCCESS
+		})
+
+		// Can be replaced by require.NoEmptyf(…) once https://github.com/stretchr/testify/pull/1481 is merged
+		if !assert.NotEmptyf(c, images, "No successful SBOM yet") {
 			return
 		}
+
+		images = lo.Filter(images, func(image *aggregator.SBOMPayload, _ int) bool {
+			cyclonedx := image.GetCyclonedx()
+			return cyclonedx != nil &&
+				cyclonedx.Metadata != nil &&
+				cyclonedx.Metadata.Component != nil
+		})
+
 		// Can be replaced by require.NoEmptyf(…) once https://github.com/stretchr/testify/pull/1481 is merged
-		if !assert.NotEmptyf(c, images, "No SBOM yet") {
+		if !assert.NotEmptyf(c, images, "No SBOM with complete CycloneDX") {
 			return
 		}
 
-		expectedTags := []*regexp.Regexp{
-			regexp.MustCompile(`^architecture:(amd|arm)64$`),
-			regexp.MustCompile(`^git\.commit\.sha:`),
-			regexp.MustCompile(`^git\.repository_url:https://github\.com/DataDog/test-infra-definitions$`),
-			regexp.MustCompile(`^image_id:ghcr\.io/datadog/apps-nginx-server@sha256:`),
-			regexp.MustCompile(`^image_name:ghcr\.io/datadog/apps-nginx-server$`),
-			regexp.MustCompile(`^image_tag:main$`),
-			regexp.MustCompile(`^os_name:linux$`),
-			regexp.MustCompile(`^short_image:apps-nginx-server$`),
+		for _, image := range images {
+			if !assert.NotNil(c, image.GetCyclonedx().Metadata.Component.Properties) {
+				continue
+			}
+
+			expectedTags := []*regexp.Regexp{
+				regexp.MustCompile(`^architecture:(amd|arm)64$`),
+				regexp.MustCompile(`^git\.commit\.sha:`),
+				regexp.MustCompile(`^git\.repository_url:https://github\.com/DataDog/test-infra-definitions$`),
+				regexp.MustCompile(`^image_id:ghcr\.io/datadog/apps-nginx-server@sha256:`),
+				regexp.MustCompile(`^image_name:ghcr\.io/datadog/apps-nginx-server$`),
+				regexp.MustCompile(`^image_tag:main$`),
+				regexp.MustCompile(`^os_name:linux$`),
+				regexp.MustCompile(`^short_image:apps-nginx-server$`),
+			}
+			err = assertTags(image.GetTags(), expectedTags)
+			assert.NoErrorf(c, err, "Tags mismatch")
+
+			properties := lo.Associate(image.GetCyclonedx().Metadata.Component.Properties, func(property *cyclonedx_v1_4.Property) (string, string) {
+				return property.Name, *property.Value
+			})
+
+			if assert.Contains(c, properties, "aquasecurity:trivy:RepoTag") {
+				assert.Equal(c, "ghcr.io/datadog/apps-nginx-server:main", properties["aquasecurity:trivy:RepoTag"])
+			}
+
+			if assert.Contains(c, properties, "aquasecurity:trivy:RepoDigest") {
+				assert.Contains(c, properties["aquasecurity:trivy:RepoDigest"], "ghcr.io/datadog/apps-nginx-server@sha256:")
```

Comment:
I guess it works here but we could have one or many RepoTag/RepoDigest

Reply:
`lo.Associate` would keep only one of the repoTag/RepoDigest.
So, the test checks that we have at least one repoTag and at least one RepoDigest.
I’m not sure that’s an issue since, anyhow, in order to have repoTag/repoDigest, we would need to pull the same image with several different names. And in the e2e tests, the `apps-nginx-server` image is pulled only with a single name.

So, in the tests, I think we can have only a single repoTag/repoDigest.
But anyhow, the test is checking that we have at least one. So that, if we have several, the test would pass.